src/models/user.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * User Model
+ */
+import * as Knex from 'knex';
+import { omit, filter, pick, invertObj } from 'ramda';
+import { Dictionary, Map } from '../types/internal';
+import { User, UserRow, UserCollection, QueryOptions } from './models';
+import { applyQueryModifiers } from './util';
+
+/*
+ * Field name mappings
+ *
+ * ColumnToModel - DB column names       -> keys of the User type
+ * ModelToColumn - keys of the User type -> DB column names
+ */
+const ColumnToModel: Map<keyof UserRow, keyof User> = {
+  user_account_id: 'id',
+  user_name: 'name',
+  user_password: 'password',
+  email: 'email',
+  qr_code: 'qrCode',
+  birth_year: 'birthYear',
+  post_code: 'postCode',
+  is_email_confirmed: 'isEmailConfirmed',
+  is_phone_number_confirmed: 'isPhoneNumberConfirmed',
+  is_email_contact_consent_granted: 'isEmailConsentGranted',
+  is_sms_contact_consent_granted: 'isSMSConsentGranted',
+  created_at: 'createdAt',
+  modified_at: 'modifiedAt',
+  deleted_at: 'deletedAt',
+  'gender.gender_name': 'gender',
+  'ethnicity.ethnicity_name': 'ethnicity',
+  'disability.disability_name': 'disability',
+};
+const ModelToColumn = invertObj(ColumnToModel);
+
+/*
+ * Implementation of the UserCollection type
+ */
+export const Users: UserCollection = {
```

Comment:
We're both using very different naming conventions. I think we need to pick a naming convention for this project as matter of priority. It's going to be a lot of unnecessary work doing this later down the line, and will complicate code throughout the app until this point.

Reply:
Are you talking about `Users` or `UserCollection`? The term "collection" comes from ORM-like libraries (e.g. Bookshelf) that use the term to describe objects that represent the entire set of a particular type of entity. I think I could have alternatively used "repository" which is used for the same thing in Laravel (PHP).

If you mean there's an inconsistency between this usage and the usage of "interface" in the roles/permissions objects, I suppose it depends on how whether you consider those objects to be model objects (i.e. something that is supposed to represent an entity in a way that hides how that entity is actually implemented in the database), or not.

The API they have at the moment is not consistent with the "Collection" interface defined in `models.d.ts`, and I'm not sure they need to be.

Maybe we can talk about this on hangouts, I don't think it can be easily resolved here.